### DISCUSSION TITLE
Duck DB CLI as Linux Service??

### DISCUSSION THREAD

**Discussion by karthikmallireddy** [duckdb/duckdb#2680] [Q&A]
Is it possible to create Duck DB CLI as Linux Service?? If yes Please share the steps

**Comment by Mytherin** [comment]
DuckDB is an embedded database and does not generally support the client/server model currently (outside of the REST server). Running the CLI as a service that runs in the background does not make much sense.

**Comment by mskyttner** [comment]
Here are some steps for running duckdb_rest_server and duckdb CLI as containerized services:

https://github.com/mskyttner/duckdbrest

The REST sever frontend is a bit tied to the schema in the TPCH database so that database is generated and embedded in the container image. If one tries to change to using another database, one will currently need to adjust the frontend code. It would be quite nice if the frontend would pick up its callback URL from wherever it was running and if the hints/tooltips data would come from the current database used. 

The duckdb CLI is webified using [`gotty`](https://github.com/yudai/gotty) which can be used to turn a CLI utility into a web application. Perhaps it could be interesting as a demo, but there is no REST API to the data provided with this variant.
